### PR TITLE
added try/except for cannot start a transaction within a transaction

### PR DESCRIPTION
**Pull request by wjblanke**

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -819,11 +819,17 @@ async def delete_unconfirmed_transactions(self, request):
...
-                    self.service.wallet_state_manager.wallets[wallet_id].target_state = None
-                await self.service.wallet_state_manager.tx_store.db_wrapper.commit_transaction()
+                try:
+                    await self.service.wallet_state_manager.tx_store.db_wrapper.begin_transaction()
+                    await self.service.wallet_state_manager.tx_store.delete_unconfirmed_transactions(wallet_id)
+                    if self.service.wallet_state_manager.wallets[wallet_id].type() == WalletType.POOLING_WALLET.value:
+                        self.service.wallet_state_manager.wallets[wallet_id].target_state = None
+                    await self.service.wallet_state_manager.tx_store.db_wrapper.commit_transaction()
+                except Exception:
+                    self.log.exception("Exception while delete_unconfirmed_transactions")
```
Comment (altendky) [inline chia/rpc/wallet_rpc_api.py:829]:
```suggestion
                    log.exception("Exception while delete_unconfirmed_transactions")
```

**Review by altendky — CHANGES_REQUESTED** [review]
pylint found us a couple things.

**Comment by wjblanke** [comment]
oof 
https://github.com/Chia-Network/chia-blockchain/runs/6884329148?check_suite_focus=true#step:10:81

**Comment by wjblanke** [comment]
maybe this needs a lock too
https://github.com/Chia-Network/chia-blockchain/blame/long_lived/atari/tests/pools/test_wallet_pool_store.py#L47

**Review by altendky — DISMISSED** [review]

**Pull request closed without merging**